ledger/acctupdates_test.go

```diff
@@ -1322,6 +1322,77 @@ func TestBoxNamesByAppIDs(t *testing.T) {
 	}
 }
 
+func TestBoxKVNoCache(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	ledgertesting.TestConsensusRange(t, boxVersion, 0, func(t *testing.T, ver int, cv protocol.ConsensusVersion, cfg config.Local) {
+		initialBlocksCount := 1
+		accts := make(map[basics.Address]basics.AccountData)
+
+		protoParams := config.Consensus[cv]
+
+		ml := makeMockLedgerForTracker(t, true, initialBlocksCount, cv,
+			[]map[basics.Address]basics.AccountData{accts},
+		)
+		defer ml.Close()
+
+		au, _ := newAcctUpdates(t, ml, cfg)
+		defer au.close()
+
+		knownCreatables := make(map[basics.CreatableIndex]bool)
+		opts := auNewBlockOpts{ledgercore.AccountDeltas{}, protocol.ConsensusCurrentVersion, protoParams, knownCreatables}
+
+		currentRound := basics.Round(1)
+
+		addKV2AU := func(appID basics.AppIndex, boxKey, boxValue []byte) {
+			boxChange := ledgercore.KvValueDelta{Data: boxValue}
+			auNewBlock(t, currentRound, au, accts, opts, map[string]ledgercore.KvValueDelta{
+				apps.MakeBoxKey(uint64(appID), string(boxKey)): boxChange,
+			})
+			auCommitSync(t, currentRound, au, ml)
+
+			// ensure rounds
+			rnd := au.latest()
+			require.Equal(t, currentRound, rnd)
+			if uint64(currentRound) > cfg.MaxAcctLookback {
+				require.Equal(t, currentRound-basics.Round(cfg.MaxAcctLookback), au.cachedDBRound)
+			} else {
+				require.Equal(t, basics.Round(0), au.cachedDBRound)
+			}
+			currentRound++
+		}
+
+		readKey := func(appID basics.AppIndex, boxKey []byte) ([]byte, error) {
+			res, err := au.LookupKv(currentRound-1, apps.MakeBoxKey(uint64(appID), string(boxKey)))
+			return res, err
+		}
+
+		testcases := []struct {
+			appID    basics.AppIndex
+			boxKey   []byte
+			boxValue []byte
+		}{
+			{appID: 10, boxKey: []byte("alice"), boxValue: []byte("value-alice")},
+			{appID: 20, boxKey: []byte("bob--"), boxValue: []byte("value-bob--")},
+			{appID: 30, boxKey: []byte("carol"), boxValue: []byte("value-carol")},
+			{appID: 40, boxKey: []byte("dave-"), boxValue: []byte("value-dave-")},
+			{appID: 50, boxKey: []byte("frank"), boxValue: []byte("value-frank")},
+		}
+
+		addKV2AU(testcases[0].appID, testcases[0].boxKey, testcases[0].boxValue)
+		addKV2AU(testcases[1].appID, testcases[1].boxKey, testcases[1].boxValue)
+		addKV2AU(testcases[2].appID, testcases[2].boxKey, testcases[2].boxValue)
+		addKV2AU(testcases[3].appID, testcases[3].boxKey, testcases[3].boxValue)
+		val, err := readKey(testcases[0].appID, testcases[0].boxKey)
+		require.NoError(t, err)
+		require.Equal(t, testcases[0].boxValue, val)
+		addKV2AU(testcases[4].appID, testcases[4].boxKey, testcases[4].boxValue)
+		_, err = readKey(testcases[0].appID, testcases[0].boxKey)
+		require.NoError(t, err)
```

Comment:
this is the place that will hit error, for first box (`testcase[0]`) goes into db after kvdelta:
- with LRU cache on, account-update reads from cache and still can get result, so it will not hit db read.
- otherwise, with nothing read from LRU cache and kvdelta, goes into DB and trigger panic.

Reply:
great. thanks for this sanity check.